plugin-build/src/main/kotlin/io/sentry/android/gradle/instrumentation/util/CatchingMethodVisitor.kt

```diff
@@ -0,0 +1,36 @@
+package io.sentry.android.gradle.instrumentation.util
+
+import io.sentry.android.gradle.SentryPlugin
+import io.sentry.android.gradle.instrumentation.MethodContext
+import io.sentry.android.gradle.util.error
+import org.objectweb.asm.MethodVisitor
+import org.slf4j.Logger
+
+interface ExceptionHandler {
+    fun handle(exception: Exception)
+}
+
+class CatchingMethodVisitor(
+    apiVersion: Int,
+    prevVisitor: MethodVisitor,
+    private val className: String,
+    private val methodContext: MethodContext,
+    private val exceptionHandler: ExceptionHandler? = null,
+    private val logger: Logger = SentryPlugin.logger
+) : MethodVisitor(apiVersion, prevVisitor) {
+
+    override fun visitMaxs(maxStack: Int, maxLocals: Int) {
+        try {
+            super.visitMaxs(maxStack, maxLocals)
+        } catch (e: Exception) {
```

Comment:
```suggestion
        } catch (e: Throwable) {
```

also check in the other catch blocks, always `Throwable` instead of `Exception`